app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/Delete.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace  Magento\Inventory\Block\Adminhtml\Edit\Button;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface;
+use Magento\Backend\Block\Context;
+use Magento\Framework\UrlInterface;
+
+/**
+ * Class Delete
+ * Configures "Delete" button on the Source Management edit form.
+ */
+class Delete implements ButtonProviderInterface
+{
+    /**
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var SourceRepositoryInterface
+     */
+    private $sourceRepository;
+
+    /**
+     * @param UrlInterface $urlBuilder
+     * @param SourceRepositoryInterface $sourceRepository
+     * @param Context $context
+     */
+    public function __construct(
+        UrlInterface $urlBuilder,
+        SourceRepositoryInterface $sourceRepository,
+        Context $context
+    ) {
+        $this->urlBuilder = $urlBuilder;
+        $this->sourceRepository = $sourceRepository;
+        $this->context = $context;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getButtonData()
+    {
+        $data = [];
+        // This part should be refactored. Implemented for test purposes.
```

Comment:
* There is no reason to get Source by Id from SourceRepository, since delete controller will validate it anyways.
* We should research building delete url on frontend, to avoid getting Request in block, which is very bad.